explainer.md

```diff
@@ -42,53 +47,85 @@ On a system that does not use file extensions but associates files with MIME typ
 
 The user can right click on CSV or SVG files in the operating system's file browser, and choose to open the files with the Grafr web application. (This option would only be presented if Grafr has been [installed](https://w3c.github.io/manifest/#installable-web-applications).)
 
-A LaunchEvent containing a sequence<[FileSystemFileHandle](https://github.com/WICG/writable-files/blob/master/EXPLAINER.md)> is then sent to the service worker, allowing the web application to decide where to open the files (i.e. in a new or existing client).
+This would create a new top level browsing context, navigating to '{origin}{action}?name={HANDLER_NAME}'. Assuming the user opened `graph.csv` in Graphr the url would be `https://graphr.com/open-files/?name=raw`. When the `load` event is fired, an additional `launchParams` property will be available on the event, containing a list of the files that the application was launched with.
 
-```js
-  self.addEventListener('launch', event => {
-    event.waitUntil(async () => {
-      const allClients = await clients.matchAll();
-      // If there isn't one available, open a window.
-      if (allClients.length === 0) {
-        const client = clients.openWindow('/');
-        client.postMessage(event.files);
-        client.focus();
-        return;
-      }
+> Note: Possibly we could use `DOMContentLoaded` or create a custom event type instead of using `load`.
 
-      const client = allClients[0];
-      client.postMessage(event.files);
-      client.focus();
-    }());
-  });
+The shape of `LoadEvent` and `LaunchParams` is described below:
+```cs
+interface LaunchParams {
+  // Cause of the launch (e.g. files|share|shortcut|link). Only files will be supported initially but will likely be added in future.
+  readonly attribute DOMString cause;
+  // The files the application was launched with. 
+  sequence<FileSystemFileHandle>? files;
+}
+
+interface LoadEvent : Event {
```

Comment:
nit: Perhaps add a disclaimer that the load event might not be the perfect place for this but we're looking for something conceptually similar.